### ISSUE TITLE
Ubuntu setup fails on Ansible/Docker 2.2.1 setup with "Cannot uninstall 'urllib3'"

### ISSUE THREAD

**Issue by nerdguru** [apache/incubator-openwhisk#3605]
## Environment details:

Used Vagrant to launch an AMI in AWS (both ami-af868e9f in US West 2 and ami-019abc64 in US East 2 reproduce the error).  

## Steps to reproduce the issue:

1.   Used vagrant to create a VM in AWS
2.   'vagrant ssh' into VM
3.  Installed git with:
sudo add-apt-repository ppa:git-core/ppa
  sudo apt-get update
  sudo apt-get install git -y

3.   Followed instructions here, after the git install line: https://github.com/apache/incubator-openwhisk/blob/master/tools/ubuntu-setup/README.md

## Provide the expected results and outputs:

Expected installation to proceed cleanly.

## Provide the actual results and outputs:
Within the ansible.sh installation of Docker 2.2.1, the following occurs:
```
+ sudo pip install docker==2.2.1
The directory '/home/ubuntu/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/ubuntu/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting docker==2.2.1
/usr/local/lib/python2.7/dist-packages/pip/_vendor/urllib3/util/ssl_.py:339: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/d9/af/a02ebab0c9791f66addb7e460dc3085240c4ec2c8f33c146fddc18156968/docker-2.2.1-py2.py3-none-any.whl (107kB)
    100% |████████████████████████████████| 112kB 3.8MB/s 
Collecting backports.ssl-match-hostname>=3.5; python_version < "3.5" (from docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/76/21/2dc61178a2038a5cb35d14b61467c6ac632791ed05131dda72c20e7b9e23/backports.ssl_match_hostname-3.5.0.1.tar.gz
Requirement already satisfied: six>=1.4.0 in /usr/lib/python2.7/dist-packages (from docker==2.2.1) (1.5.2)
Collecting websocket-client>=0.32.0 (from docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/9d/fb/f51a03e232e00d6c504dfe815aed090c894ba3f8d3f7fd9612f3e227bf24/websocket_client-0.47.0-py2.py3-none-any.whl (200kB)
    100% |████████████████████████████████| 204kB 7.2MB/s 
Requirement already satisfied: ipaddress>=1.0.16; python_version < "3.3" in /usr/local/lib/python2.7/dist-packages (from docker==2.2.1) (1.0.22)
Collecting docker-pycreds>=0.2.1 (from docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/5c/b6/3a4ca03225f6e9a25cb47cd3c695d19b88df5928ab9310e64dec1eb3b4d1/docker_pycreds-0.2.3-py2.py3-none-any.whl
Collecting requests!=2.11.0,!=2.12.2,>=2.5.2 (from docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/49/df/50aa1999ab9bde74656c2919d9c0c085fd2b3775fd3eca826012bef76d8c/requests-2.18.4-py2.py3-none-any.whl (88kB)
    100% |████████████████████████████████| 92kB 9.1MB/s 
Collecting urllib3<1.23,>=1.21.1 (from requests!=2.11.0,!=2.12.2,>=2.5.2->docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/63/cb/6965947c13a94236f6d4b8223e21beb4d576dc72e8130bd7880f600839b8/urllib3-1.22-py2.py3-none-any.whl (132kB)
    100% |████████████████████████████████| 133kB 7.6MB/s 
Requirement already satisfied: idna<2.7,>=2.5 in /usr/local/lib/python2.7/dist-packages (from requests!=2.11.0,!=2.12.2,>=2.5.2->docker==2.2.1) (2.6)
Collecting chardet<3.1.0,>=3.0.2 (from requests!=2.11.0,!=2.12.2,>=2.5.2->docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 9.1MB/s 
Collecting certifi>=2017.4.17 (from requests!=2.11.0,!=2.12.2,>=2.5.2->docker==2.2.1)
  Downloading https://files.pythonhosted.org/packages/7c/e6/92ad559b7192d846975fc916b65f667c7b8c3a32bea7372340bfe9a15fa5/certifi-2018.4.16-py2.py3-none-any.whl (150kB)
    100% |████████████████████████████████| 153kB 9.8MB/s 
cheetah 2.4.4 requires Markdown>=2.0.1, which is not installed.
Installing collected packages: backports.ssl-match-hostname, websocket-client, docker-pycreds, urllib3, chardet, certifi, requests, docker
  Running setup.py install for backports.ssl-match-hostname ... done
  Found existing installation: urllib3 1.7.1
Cannot uninstall 'urllib3'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
/usr/local/lib/python2.7/dist-packages/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
```

## Additional information you deem important:
* Might this be an AMI issue I'm having or some Python version conflict as the error indicates or something else?

**Comment by csantanapr** [comment]
What is AMI?

**Comment by nerdguru** [comment]
Amazon Machine Image.

ami-af868e9f in US West 2 can be found here, as an example:
http://cloud-images.ubuntu.com/releases/14.04.3/14.04.3/

The 64-bit HVM.

**Comment by csantanapr** [comment]
Thanks @nerdguru for reporting the issue

**Comment by csantanapr** [comment]
@nerdguru 
I was able to reproduce locally with a fresh install of ubuntu14.04.05 on virtual boxe

There is a problem with some pip packages, I have a proposed fix in #3614

Give it a try to see if that helps.

**Comment by nerdguru** [comment]
Confirmed @csantanapr that works now. Thanks so much!
